src/Microsoft.Azure.WebJobs.Logging/Internal/LogWriter.cs

```diff
@@ -152,6 +156,15 @@ public Task AddAsync(FunctionInstanceLogItem item, CancellationToken cancellatio
             // Both Start and Completed log here. Completed will overwrite a Start entry. 
             lock (_lock)
             {
+                // Permanent failures when flushing to storage can result in many log entries being buffered.
+                // This basically becomes a memory leak. Mitigate by enforcing a max.
+                if (_activeFuncs.Count >= MaxBufferedEntryCount || _completedFunctions.Count >= MaxBufferedEntryCount)
```

Comment:
Should this be comparing `_activeFuncs.Count + _completedFunctions.Count >= MaxBufferedEntryCount`? Otherwise it seems like you could approach 2x that number of queued functions.

Reply:
I had it this way originally and changed it because a given log entry is often written to both of these, which made things confusing because you would hit the buffer limit twice as fast as expected. I think its more intuitive this way.